src/Common/src/Interop/Unix/System.Globalization.Native/Interop.Idna.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class GlobalizationNative
+    {
+        internal const int AllowUnassigned = 0x1;
+        internal const int UseStd3AsciiRules = 0x2;
+
+        [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode)]
+        internal static extern int ToAscii(uint flags, string src, int srcLen, [Out] char[] dstBuffer, int dstBufferCapacity);
+
+        [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode)]
+        internal static extern int ToUnicode(uint flags, string src, int srcLen, [Out] char[] dstBuffer, int dstBufferCapacity);
```

Comment:
With MCG, there may be a slight perf benefit to specifying `[Out]`. Not sure if this applies to the CoreCLR marshaller.

From http://blogs.msdn.com/b/dotnet/archive/2014/06/13/net-native-deep-dive-debugging-into-interop-code.aspx:

> However, if you look very closely, there is actually a step that is a bit unnecessary: MCG copies the contents of the StringBuilder into the buffer, while the API doesn't expect you to pass in any string value at all - it simply fill in the buffer. The copying step is actually wasted. Fortunately, .NET allows you to customize the marshaling behavior by specifying [In]/[Out] attribute on the parameter (for more information, you can refer to this MSDN article). StringBuilder by default gets [In, Out], and in this case, we only want the [Out] direction because we only need to receive data from the LCIDToLocaleName API